File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        int[] arr = new int[n+1];
        for(int num : nums) arr[num]++;
        List<Integer> list = new ArrayList<>();
        for(int i=1; i<=n; i++){
            if(arr[i]==0) list.add(i);
        }
        return list;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        for(int num : nums){
            if(num<0) num*=-1;
            if(nums[num-1]>0) nums[num-1]*=-1;
        }
        vector<int> vec;
        for(int i=0; i<n; i++){
            if(nums[i]>0) vec.push_back(i+1);
        }
        return vec;
    }
};